backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java

```diff
@@ -95,14 +97,62 @@ protected String getUriOfAttachment(AttachmentContent attachmentContent) throws
         return new URI("file", filePath, null).toString();
     }
 
+    protected Stream<String> getAllLicenseTextsFromInfo(AnyLicenseInfo spdxLicenseInfo) {
+        if (spdxLicenseInfo instanceof LicenseSet) {
+
+            LicenseSet LicenseSet = (LicenseSet) spdxLicenseInfo;
+            return Arrays.stream(LicenseSet.getMembers())
+                    .flatMap(this::getAllLicenseTextsFromInfo);
+
+        } else if (spdxLicenseInfo instanceof ExtractedLicenseInfo) {
+
+            ExtractedLicenseInfo extractedLicenseInfo = (ExtractedLicenseInfo) spdxLicenseInfo;
+            return Collections.singleton(extractedLicenseInfo.getExtractedText())
+                    .stream();
+
+        } else if (spdxLicenseInfo instanceof License) {
+
+            License license = (License) spdxLicenseInfo;
+            return Collections.singleton(license.getLicenseText())
+                    .stream();
+
+        } else if (spdxLicenseInfo instanceof OrLaterOperator) {
+
+            OrLaterOperator orLaterOperator = (OrLaterOperator) spdxLicenseInfo;
+            return getAllLicenseTextsFromInfo(orLaterOperator.getLicense());
+
+        } else if (spdxLicenseInfo instanceof WithExceptionOperator) {
+
+            WithExceptionOperator withExceptionOperator = (WithExceptionOperator) spdxLicenseInfo;
+            String licenseExceptionText = withExceptionOperator.getException()
+                    .getLicenseExceptionText();
+            return getAllLicenseTextsFromInfo(withExceptionOperator.getLicense())
+                    .map(licenseText -> licenseText + "\n\n" + licenseExceptionText);
```

Comment:
The license text, of an license with exception is set to

```
${licenseText}

${licenseExceptionText}
```

which might be redundant or/and insufficient. This might also need some clarification.